mautic/mautic
# Fixed batch delete response inconsistencies

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? | y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This fixes the issue with inconsistent response formats when using the batch entity delete endpoint. It basically ensures that if the ID list from the request is keyed as an associative array, it'll return a json object response. If it's a simple array (sequential numeric array that starts with 0), it'll return a json array.

Yes, the implementation to start with is undesirable but this fixes the issue without introducing a BC break nor rewriting our API ;-) 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Hit up the /api/contacts/batch/delete end a few times with different IDs
2. Note the response that sometimes the `contacts` array is json array and that sometimes it's a json object (keyed by a number)
3. Run the new unit tests

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Repeat and the response should be consistent
